src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -453,6 +460,22 @@ public void process(HttpMessage msg) {
 			Stats.clear(this.getParam(params, PARAM_KEY_PREFIX, ""));
 			return ApiResponseElement.OK;
 			
+		} else if (ACTION_DELETE_SITE_NODE.equals(name)) {
+			try {
+				String url = params.getString(PARAM_URL);
+				String method = getParam(params, PARAM_METHOD, "GET");
+				String postData = params.getString(PARAM_POST_DATA);
+				URI uri = new URI(url, true);
+				SiteMap siteMap = session.getSiteTree();
+				SiteNode siteNode = siteMap.findNode(uri, method, postData);
+				if(siteNode == null) {
+					throw new ApiException(ApiException.Type.ILLEGAL_PARAMETER, PARAM_URL);
```

Comment:
`DOES_NOT_EXIST` instead of `ILLEGAL_PARAMETER` might describe better this case.